src/utils/getActualDataList.ts

```diff
@@ -0,0 +1,31 @@
+import dayjs, { extend } from 'dayjs';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+
+type DataListType<T> = Array<T>;
+type DataItem = Record<string, string | HTMLElement>;
+interface DataItemWithDate extends DataItem {
+  date: string; // 'YYYY-MM-DD'
+}
+
+export type ActualDataListProps = {
+  dataList: DataListType<DataItemWithDate>;
+  actualDelayInDays: number;
+  sorted?: boolean;
+};
+
+export const getActualDataList = ({
+  dataList,
+  actualDelayInDays,
+  sorted = true,
+}: ActualDataListProps): DataListType<DataItemWithDate> | [] => {
+  extend(customParseFormat);
```

Comment:
What about to add generic to the function, in order to prevent type casting like this
![image](https://github.com/rolling-scopes/site/assets/114234698/3468bdaa-866f-43f3-a739-dce516af089a)

Reply:
This function is universal for all such lists - e.g. events, courses, etc. Therefore, I carry out typing at the stage of its use.

Reply:
I got you, but in order to prevent type casting we can change this function to utilize generics and carry out typing return value at function call stage

```tsx
const actualEvents = getActualDataList<AnyTypeYouWantToReturn>({
  dataList: events,
  actualDelayInDays: 3,
});
```